src/main/flight/mixer.c

```diff
@@ -759,6 +763,8 @@ void mixTable(void)
             axisPID[PITCH] * currentMixer[i].pitch +
             axisPID[ROLL] * currentMixer[i].roll +
             -mixerConfig->yaw_motor_direction * axisPID[YAW] * currentMixer[i].yaw;
+
+        if (batteryConfig->vbatPidCompensation) motor[i] *= vbatCompensationFactor;  // Add voltage compensation
```

Comment:
I'm just moving my comment from the outdated diff to here so it isn't missed...

Re: you should use the form: motor[i] = throttle_zero + (motor[i]-throttle_zero)*vbatCompensation

> But mincommand is not relevant....right? It is never used. The rcCommand throttle starts from min_check and never gets lower than that.

Well it depends on the ESC calibration, and how it interprets it. I believe the ESCs are calibrated from minCommand up to maxThrottle. SimonK then takes this range and maps it to a deadband followed by a power band. minThrottle is generally set to some point above the start of the power band such that the prop reliably spins.

Now forgetting about the picky-ness of where the power=0 point is, you are applying a scale factor to something that doesn't pass through zero. Lets say motor[i] has the range of 1000 to 2000 to use round numbers - where 1000 is throttle 0% and 2000 is 100%. Your battery depletes and you get a compensation factor of 1.2, so [1000:2000] * 1.2 = [**1200**:2400] <- whoops you just scaled the zero point. What you really wanted was [**1000**:2200].

Make sense?